dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java

```diff
@@ -79,22 +88,124 @@ public void setupPerTest() throws Exception {
 
 
     /**
-     *  
-     * Skipped as this is not yet working in 4.1.0-RC1
+     *  Not HTML5, so IDs should NOT be rendered
      *
      * @throws Exception
      */
-    // @Test
-    public void verifyIDsAreGeneratedOnHead() throws Exception {
+    @Test
+    public void verifyIDsAreGenerated_WithIDs_HTML4() throws Exception {
+        try (WebClient webClient = new WebClient()) {
+            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "HTML4_withIDs.xhtml_xhtml");
+
+            HtmlPage page = (HtmlPage) webClient.getPage(url);
+
+            logPage(page);
+
+            assertEquals("Id attribute not found on head!", "", page.getHead().getId());
+            assertEquals("Id attribute not found on head!", "", page.getBody().getId());
+        }
+    }
+
+    /*
+     *  HTML5 output and IDs specified in the page, so they should be rendered. 
+     * 
+     *  This is the only test where are expeced to appear!
+     * 
+     *  @throws Exception
+     */
+    @Test
+    public void verifyIDsAreGenerated_WithIDs_HTML5() throws Exception {
         try (WebClient webClient = new WebClient()) {
-            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "index.xhtml");
+            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "HTML5_withIDs.xhtml");
 
             HtmlPage page = (HtmlPage) webClient.getPage(url);
 
-            // assertNotNull("Id attribute not found on head!", page.getHead().getId());
+            logPage(page);
+
+            assertEquals("Id attribute was found on head!", "headID", page.getHead().getId());
```

Comment:
Shouldn't the assert message be something like "Id attribute was not found on head!"?

Reply:
Please review your messages in this class as I think they need some work I won't comment on everyone.